pkg/cluster/backup_manager.go

```diff
@@ -56,16 +56,16 @@ func (bm *backupManager) setupStorage() (s backupstorage.Storage, err error) {
 	b := cl.Spec.Backup
 	switch b.StorageType {
 	case spec.BackupStorageTypePersistentVolume, spec.BackupStorageTypeDefault:
-		s, err = backupstorage.NewPVStorage(c.KubeCli, cl.Name, cl.Namespace, c.PVProvisioner, *b)
+		s, err = backupstorage.NewPVStorage(c.KubeCli, cl.Metadata.Name, cl.Metadata.Namespace, c.PVProvisioner, *b)
```

Comment:
See https://github.com/wfarr/k8s-tpr-playground/pull/1/files

Basically, the object inherits `CodecDecodeSelf()` from `ObjectMeta` .

I didn't know much about UnmarshalJSON() workaround either. That's why I also mentioned the alternative solution. It's clearer, and better understood.